### PR TITLE
[Workflow] Clearer `NotEnabledTransitionException` message

### PR DESCRIPTION
**Pull request by that-guy-iain**
| Q             | A
| ------------- | ---
| Branch?       | 7.2 
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| License       | MIT

The current exception message is somewhat confusing as it makes you think the transition is not enabled at all within the workflow. This can/does lead to people trying to debug a config issue. While the problem being encountered issue is that the transition can't be applied due to the current state of the supported class/object.

Hopefully, this makes it a bit clearer.

### CODE REVIEW

File: src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
 use Symfony\Component\Workflow\WorkflowInterface;
 
 /**
- * Thrown by Workflow when a not enabled transition is applied on a subject.
+ * Thrown by Workflow when a not enabled transition is applable on a subject.
```
Comment (OskarStark) [inline src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php]: typo?

**Review by lyrixx — COMMENTED** [review]
I'm not sure it's better. But I'm not against it.
I let other folks decide about this one.

For the record, the word `enabled` comes from the pretinet vocabulary

https://en.wikipedia.org/wiki/Petri_net#:~:text=enabled

**Comment by that-guy-iain** [comment]
@lyrixx Why do you think it's not better?

The reason I think the current one is confusing and misleading is from a conversation on the slack's support channel https://symfony-devs.slack.com/archives/C3EQ7S3MJ/p1720656379828679 where it was pretty confusing and a guessing game until I dug into the code.

**Comment by OskarStark** [comment]
As Slack deletes conversations, I screenshotted the conversation:
CleanShot 2024-07-12 at 12 27 50@2x

File: src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php [hunk 2]
```diff
@@ -28,7 +28,7 @@ public function __construct(
         private TransitionBlockerList $transitionBlockerList,
         array $context = [],
     ) {
-        parent::__construct($subject, $transitionName, $workflow, \sprintf('Transition "%s" is not enabled for workflow "%s".', $transitionName, $workflow->getName()), $context);
+        parent::__construct($subject, $transitionName, $workflow, \sprintf('Can not apply transition to "%s" for workflow "%s".', $transitionName, $workflow->getName()), $context);
```
Comment (OskarStark) [inline src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php]: the first parameter is the name of the transition, so the sentence is confusing:
`Can not apply transition to "transitionName" ...?`

Or am I missing sth.?

You always use a transition to get to a place, not to another transition

File: src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php [hunk 3]
```diff
@@ -15,7 +15,7 @@
 use Symfony\Component\Workflow\WorkflowInterface;
 
 /**
- * Thrown by Workflow when a not enabled transition is applied on a subject.
+ * Thrown by Workflow when a not enabled transition is appliable on a subject.
```
Comment (xabbuh) [inline src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php]: What is the reason for this change?

File: src/Symfony/Component/Workflow/Tests/WorkflowTest.php [hunk 4]
```diff
@@ -286,7 +286,7 @@ public function testApplyWithNotEnabledTransition()
 
             $this->fail('Should throw an exception');
         } catch (NotEnabledTransitionException $e) {
-            $this->assertSame('Transition "t2" is not enabled for workflow "unnamed".', $e->getMessage());
+            $this->assertSame('Can not apply transition to "t2" for workflow "unnamed".', $e->getMessage());
```
Comment (xabbuh) [inline src/Symfony/Component/Workflow/Tests/WorkflowTest.php]: IMO the change to this test shows that the change is not correct. t2 is a transition, right?

File: src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php (hunk 2 above)
Comment (that-guy-iain) [reply src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php:31]: So changer to can not apply transition of %s ?

File: src/Symfony/Component/Workflow/Tests/WorkflowTest.php (hunk 4 above)
Comment (xabbuh) [reply src/Symfony/Component/Workflow/Tests/WorkflowTest.php]: My comment was about the assertion per se but what we expect now does not seem logical. Why do we expect a message where the name of the transaition is used in a way that makes it sound that we try to reach a place named the same as the transition?

File: src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php (hunk 2 above)
Comment (OskarStark) [reply src/Symfony/Component/Workflow/Exception/NotEnabledTransitionException.php]: "Can not apply transition "%s" for workflow."

**Review by xabbuh — APPROVED** [review]

**Review by lyrixx — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @that-guy-iain.
